lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -9,19 +9,31 @@ class FirebaseAppDistributionApiClient
       MAX_POLLING_RETRIES = 60
       POLLING_INTERVAL_SECONDS = 2
 
+      # Enables tester access to the specified app release. Skips this
+      # step if no testers are passed in (emails and group_ids are nil).
+      #
+      # args
+      #   app_id - Firebase App ID
+      #   release_id - App release ID, returned by upload_status endpoint
+      #   emails - String array of app testers' email addresses
+      #   group_ids - String array of Firebase tester group IDs
+      #
+      # Throws a user_error if app_id, emails, or group_ids are invalid
       def enable_access(app_id, release_id, emails, group_ids)
         if emails.nil? && group_ids.nil?
           UI.message("No testers passed in. Skipping this step")
           return
         end
-        begin
         payload = { emails: emails, groupIds: group_ids }
-        connection.post(enable_access_url(app_id, release_id), payload.to_json) do |request|
-          request.headers["Authorization"] = "Bearer " + auth_token
+        begin
+          connection.post(enable_access_url(app_id, release_id), payload.to_json) do |request|
+            request.headers["Authorization"] = "Bearer " + auth_token
+          end
+        rescue Faraday::ResourceNotFound
+          UI.user_error!("#{ErrorMessage::INVALID_APP_ID}: #{app_id}")
```

Comment:
This change makes me realize that these should probably `.crash!` instead of user_error. It's weird to have it print an error and then immediately afterwards print the success message on line 37.

If you make that change please update the doc comment as well.

Reply:
Oh wait, do these actually cause a crash? Your tests make it seem like they do. If so please disregard my comment.

Reply:
Technically this can fail if the app or release isn't found. I think the error message should be something like:
```
App distribution failed to fetch release with id <release_id> 
```

Reply:
@lfkellogg user_error does cause a crash. @tagboola I added this functionality to the post_notes pr but I can't to this method because an invalid release_id returns a client error just like invalid emails/goup_ids. How should we handle it in this case? I think that since the user should not be specifying the release_id (since the plugin just gets that from upload_status), it's not as helpful to catch this and throw it as a user_error. 
However, I do think that an invalid release_id would be a reasonable exception to catch -  will the backend eventually have functionality to differentiate between invalid release_ids vs emails/groupids?

Reply:
Ahh okay. So an invalid release_id doesn't return a 404? If that's the case I'm fine with leaving it the way it is 🙂